ironfish/src/network/peerNetwork.ts

```diff
@@ -1279,6 +1280,8 @@ export class PeerNetwork {
     const received = new Date()
     const hash = transaction.hash()
 
+    this.onTransactionGossipReceived.emit(transaction)
+
```

Comment:
The block gossip equivalent gets called further down after we've marked the block as received and filtered for duplicates, should we align them one way or the other?

Reply:
Yes I think we should move this down to after filtering for duplicates. That should remove the need for the LRU in the RPC endpoint, too